docs/reference/tcpmappings.md

```diff
@@ -0,0 +1,111 @@
+
+| Attribute                 | Description               |
+| :------------------------ | :------------------------ |
+| `address`         | (optional) the IP address on which Ambassador should listen for connections for this Mapping -- if not present, Ambassador will listen on all addresses )
+| `port`            | (required) the TCP port on which Ambassador should listen for connections for this Mapping |
+| `idle_timeout_ms` | (optional) the timeout, in milliseconds, after which the connection will be terminated if no traffic is seen -- if not present, no timeout is applied |
+| `enable_ipv4` | (optional) if true, enables IPv4 DNS lookups for this mapping's service (the default is set by the [Ambassador module](/reference/modules)) |
+| `enable_ipv6` | (optional) if true, enables IPv6 DNS lookups for this mapping's service (the default is set by the [Ambassador module](/reference/modules)) |
+
+If both `enable_ipv4` and `enable_ipv6` are set, Ambassador will prefer IPv6 to IPv4. See the [Ambassador module](/reference/modules) documentation for more information.
+
+Ambassador can manage TCP connections using TLS:
+
+| Attribute                 | Description               |
+| :------------------------ | :------------------------ |
+| [`tls`](#using-tls)       | (optional) enables TLS, and specifies the name of the `TLSContext` which will determine the certificate to offer when a client connects |
+| [`host`](/reference/host) | (optional) specifies the hostname that must be presented using SNI for this `TCPMapping` to match -- **ONLY AVAILABLE WHEN USING TLS** |
+
+Note that **Ambassador will terminate the TLS connection** when using TLS with a `TCPMapping`. You can use a `TCPMapping` to pass a TLS connection to an upstream service without terminating it, but `host` matching will not be available.
+
+Ambassador supports multiple deployment patterns for your services. These patterns are designed to let you safely release new versions of your service, while minimizing its impact on production users.
+
+| Attribute                 | Description               |
+| :------------------------ | :------------------------ |
+| [`weight`](/reference/canary)        | (optional) specifies the (integer) percentage of traffic for this resource that will be routed using this mapping |
+
+The name of the mapping must be unique.
+
+## Example `TCPMapping`s
+
+`TCPMapping` definitions are fairly straightforward. Here's an example that simply relays any connection on port 2222 onward to a Kubernetes service:
+
+```yaml
+---
+apiVersion: ambassador/v1
+kind: TCPMapping
+name: qotm_mapping
+port: 2222
+service: qotm
+```
+
+Here's the same service, with Ambassador terminating TLS:
+
+```yaml
+---
+apiVersion: ambassador/v1
+kind: TCPMapping
+name: quote_mapping
+port: 2222
+tls: some-context
+service: qotm
+```
+
+Note that in this example, Ambassador will contact the `qotm` service using cleartext HTTP, not HTTPS. You can explicitly ask Ambassador to use HTTPS:
```

Comment:
in this example above (we should specify if it's the example above or below)

Reply:
Will fix -- actually, I need to clean up the docs around ports too.